### PR TITLE
Always get blocks for articles from CAPI

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

Make sure we always get blocks for articles in the capi model, so the render tier picker can function properly.

This shouldn't change the behaviour of the existing article since we're not processing the blocks anyway. Liveblogs *do* process blocks, but they are handled differently. 

There is a chance this might mean we get marginally more data going across the wire from capi so I'll keep an eye on our performance metrics. We should see any difference in render time reflected in the rendering time cloudwatch metrics.

I also wrapped hasOnlySupportedElements in a check to make sure there are blocks before processing them just in case this function is ever called directly.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by MatthewJWalls** [comment]
I've brushed up against a problem with the articleController with this change. 

It's a bit spaghetti but basically our logic for controlling liveblog rendering is tied up with the presence of blocks (in a way that I don't think it should be). Passing through blocks in every case upsets that logic.

**Comment by MatthewJWalls** [comment]
for future reference...explanation incomming

image

**Comment by MatthewJWalls** [comment]
Note that the reason the database files have changed is because we print out some js config at the top of our pages which has  a changed revisionNumber in it. I've diff'd an article page before and after my change and that's the only change, there's no change in page content.

File: article/app/services/RenderingTierPicker.scala [hunk 1]
```diff
@@ -30,17 +30,25 @@ object RenderingTierPicker {
...
 
   private def hasOnlySupportedElements(page: PageWithStoryPackage): Boolean = {
-    ! page.article.blocks.get.body.exists((block: BodyBlock) => {
 
-      val hasUnsupportedElements: Boolean = block.elements.flatMap {
-        case b: TextBlockElement => None
-        case b: ImageBlockElement => None
-        case b => Some(b)
-      }.nonEmpty
+    page.article.blocks match {
```
Comment (QuarpT) [inline article/app/services/RenderingTierPicker.scala]: This should be simplified

File: article/app/services/RenderingTierPicker.scala [hunk 2]
```diff
@@ -30,17 +30,25 @@ object RenderingTierPicker {
...
+
+      case Some(b) => ! page.article.blocks.get.body.exists((block: BodyBlock) => {
```
Comment (QuarpT) [inline article/app/services/RenderingTierPicker.scala]: why use Option.get when you have b?

File: article/app/services/RenderingTierPicker.scala [hunk 3]
```diff
@@ -30,17 +30,25 @@ object RenderingTierPicker {
   }
 
   private def hasOnlySupportedElements(page: PageWithStoryPackage): Boolean = {
```
Comment (QuarpT) [inline article/app/services/RenderingTierPicker.scala]: how about
```
private def hasOnlySupportedElements(page: PageWithStoryPackage): Boolean = {
    def unsupportedElement(blockElement: BlockElement) = blockElement match {
      case _: TextBlockElement => true
      case _: ImageBlockElement => true
      case _ => false
    }
    !page.article.blocks.exists(_.body.exists(_.elements.exists(unsupportedElement)))
  }
```
?

Comment (MatthewJWalls) [reply article/app/services/RenderingTierPicker.scala]: Yeah that looks better, will change and test

**Review by QuarpT — APPROVED** [review]

File: article/app/controllers/ArticleController.scala [hunk 4]
```diff
@@ -142,7 +142,7 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
       Action.async { implicit request =>
 
         def renderWithRange(range: BlockRange) =
-          mapModel(path, range = Some(range)) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
+          mapModel(path, range = Some(range), asArticle=false) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
```
Comment (QuarpT) [inline article/app/controllers/ArticleController.scala]: could rename asArticle to isLiveBlog

**Comment by MatthewJWalls** [comment]
I'm getting more data than I expected so we should be able to revert this monday afternoon.

**Pull request merged**